pkgs/by-name/fa/fastjet-contrib/package.nix

```
@@ -14,13 +14,19 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-2apGVg/f2FCC8gKlqc5kdo/gxZhmDwEyBqjJwX7NDzY=";
   };
 
+  nativeBuildInputs = [
+    fastjet # fastjet-config
+  ];
```

```suggestion
  configureFlags = [
    "--fastjet-config=${lib.getExe' (lib.getDev fastjet) "fastjet-config"}"
  ];
```

i would hope similar options exist for `fltk-config` users too

---

Is there an advantage to this? We usually just include `pkg-config` rather than passing the full path to the build.

Though it does seem like it should probably be its own output or separate package.

---

if adding `pkg-config` to nativeBuildInputs and keeping `fastjet` only in buildInputs also works, then that's better, yeah (didn't try it).

i just want to make sure we _don't_ use the native `<library>-config` binaries, because they emit the _wrong_ c/ld flags (e.g. `-L /nix/store/...-native-fastjet/lib` -- not the cross compiled libraries): it's kind of a fluke that it even works, and i'm not actually sure it _does_ work for the weirder cross targets or for people using more niche stdenvs.

---

Right, okay, I see what you’re concerned about; `pkg-config` was just a comparison, not a suggestion.

But using the `hostPlatform` `fastjet-config` won’t work in cross scenarios, because the `buildPlatform` can’t necessarily execute it. So if the `fastjet-config` is tied to the `fastjet` it comes with then we have a more general problem. (Not handling cross properly is why these bespoke `*-config`s are usually evil…)